src/wp-includes/class-wp-block-bindings-registry.php

```diff
@@ -144,6 +157,25 @@ public function register( string $source_name, array $source_properties ) {
 
 		$this->sources[ $source_name ] = $source;
 
+		// Add `uses_context` defined by block bindings sources.
+		add_filter(
+			'register_block_type_args',
+			function ( $args, $block_name ) use ( $source ) {
+				$allowed_blocks = $this->allowed_blocks;
+
+				if ( empty( $allowed_blocks[ $block_name ] ) || empty( $source->uses_context ) ) {
+					return $args;
+				}
+				$original_use_context = isset( $args['uses_context'] ) ? $args['uses_context'] : array();
+				// Use array_values to reset the array keys.
+				$args['uses_context'] = array_values( array_unique( array_merge( $original_use_context, $source->uses_context ) ) );
```

Comment:
The comment included makes me feel that it needs to be covered with unit tests to ensure it works correctly.

Reply:
We might not need to call `array_values`. Every place in the code I checked leaves the array keys unmodified. What was the reason you added `array_values` call?

Reply:
I will check it again, but without `array_values` it was returning some errors caused by the array keys. If I remember correctly, when we merge two arrays like these:

```
$original_uses_context = array( 'postId' );
$source_uses_context = array( 'postId', 'postType' );
$merged_array = array_unique( array_merge( $original_use_context, source_uses_context ) );
```
The result array has an issue with the keys:
```
[0] => 'postId',
[2] => 'postType',
```
It seems to skip the repeated elements (`postId`), so I was using `array_values` to reset that.

Anyway, I will take another look at it.

Reply:
By the way, it causes an error in the editor, not the server.

Reply:
@SantosGuillamot  
You can try, a much cleaner code with the same effect.

```php
$original_uses_context = array( 'postId' );
$source_uses_context = array( 'postId', 'postType' );
$merged_array = array_unique( $original_uses_context + $source_uses_context );
```

Reply:
I've tested using `$original_uses_context + `source_uses_context`, but it doesn't work as expected.

Apart from that, I added a test where I am checking if the index is correct: [link](https://github.com/SantosGuillamot/wordpress-develop/blob/f2b3b3af8491de34530fb2802029c867a7406f93/tests/phpunit/tests/block-bindings/wpBlockBindingsRegistry.php#L357-L390).

Reply:
Let's leave it as is. We have an unit test in place that ensures that the editor doesn't crash so we can refactor that in the future safely.